src/telemetry.ts

```diff
@@ -0,0 +1,57 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as vscode from 'vscode';
+import * as util from './util';
+import TelemetryReporter from 'vscode-extension-telemetry';
+
+interface IPackageInfo {
+    name: string;
+    version: string;
+    aiKey: string;
+}
+
+let telemetryReporter: TelemetryReporter | null;
+
+export function activate(context: vscode.ExtensionContext): void {
+    try {
+        telemetryReporter = createReporter();
+        if (telemetryReporter) {
+            context.subscriptions.push(telemetryReporter);
```

Comment:
If you add this to the `context.subscriptions` you shouldn't need to manually dispose it later, right?

Reply:
I was following [this usage instruction](https://github.com/Microsoft/vscode-extension-telemetry). Probably isn't necessary?